### PR TITLE
[CM] Remove deprecated parameters for initial component states.

### PR DESCRIPTION
**Pull request by destogl**
A part of #1237

### CODE REVIEW

File: controller_manager/src/controller_manager.cpp [hunk 1]
```diff
@@ -281,10 +281,6 @@ ControllerManager::ControllerManager(
   }
   else
   {
-    RCLCPP_WARN(
-      get_logger(),
-      "[Deprecated] Passing the robot description parameter directly to the control_manager node "
-      "is deprecated. Use '~/robot_description' topic from 'robot_state_publisher' instead.");
```
Comment (destogl) [inline controller_manager/src/controller_manager.cpp:287]:
```suggestion
RCLCPP_WARN(
      get_logger(),
      "[Deprecated] Passing the robot description parameter directly to the control_manager node "
      "is deprecated. Use '~/robot_description' topic from 'robot_state_publisher' instead.");
```

File: controller_manager/src/controller_manager.cpp [hunk 2]
```diff
@@ -414,56 +414,15 @@ void ControllerManager::init_resource_manager(const std::string & robot_descript
...
-  }
-  // END: Keep old functionality on humble for backwards compatibility (Remove at the end of 2023)
-  else
-  {
-    // activate all other components
-    for (const auto & [component, state] : components_to_activate)
-    {
-      resource_manager_->set_component_state(component, active_state);
-    }
+    resource_manager_->set_component_state(component, active_state);
```
Comment (fmauch) [inline controller_manager/src/controller_manager.cpp:425]:
```suggestion
    rclcpp_lifecycle::State active_state(
      State::PRIMARY_STATE_ACTIVE, hardware_interface::lifecycle_state_names::ACTIVE);
    resource_manager_->set_component_state(component, active_state);
```

**Review by fmauch — CHANGES_REQUESTED** [review]
I could not thoroughly test this, but that compile issue is definitively there.

File: controller_manager/src/controller_manager.cpp (hunk 2 above)
Comment (destogl) [reply controller_manager/src/controller_manager.cpp:427]: Thanks!

**Review by fmauch — APPROVED** [review]
Verified that I could still successfully configure initial component state with this.

**Review by saikishor — APPROVED** [review]
LGTM

**Pull request merged**